src/weierstrass/point.rs

```diff
@@ -33,6 +34,33 @@ impl EcPoint<Scalar> for Point {
     fn is_zero(&self) -> bool {
         self.c.x.is_zero() && !self.c.y.is_zero() && self.c.z.is_zero()
     }
+    fn serialize(&self, compress: bool) -> Vec<u8> {
+        // normalize the point to ensure that z = 1
+        // clone so that we don't mutate the original point
+        let mut p_normal = self.clone();
+        p_normal.normalize();
+        let coords = &p_normal.c;
```

Comment:
It is missing the case of `P = infinity`, in such a case the output is a byte zeroed.

Reply:
The same applies for Montgomery points.